### DISCUSSION TITLE
@types/tar-stream Extract.on('entry') "next" callback should have an error parameter

### DISCUSSION THREAD

**Discussion by paya-cz** [DefinitelyTyped/DefinitelyTyped#54174] [Issues with a @types package] [answered]
Current type definition is:

```
export interface Extract extends stream.Writable {
    on(event: 'entry', listener: (headers: Headers, stream: stream.PassThrough, next: () => void) => void): this;
}
```

However, the `next` callback should accept an error parameter:
https://github.com/mafintosh/tar-stream/blob/e08fcd845744c43e1e71343f117be9993208103a/extract.js#L180
https://github.com/mafintosh/tar-stream/blob/e08fcd845744c43e1e71343f117be9993208103a/extract.js#L70

Thus, the correct type definition should be:

```
export interface Extract extends stream.Writable {
    on(event: 'entry', listener: (headers: Headers, stream: stream.PassThrough, next: (err?: any | null) => void) => void): this;
}
```

**Comment by BendingBender** [comment, marked as answer]
Fixed signature, see https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56577

**Reply by paya-cz** [reply]
Thank you!
